lib/challenge.js

```diff
@@ -58,7 +73,15 @@ var some_pet;
 //    the values of `name`, `species`, `breed`, and `noise`. The constructor function should also give each pet a method called `makeNoise`,
 //    which behaves as described in Q3.
 
-var Pet;
+var Pet = function Pet (name, species, breed, noise) {
+  this.name = name;
+  this.species = species;
+  this.breed = breed;
+  this.noise = noise;
+  this.makeNoise = [];
+
+}
```

Comment:
This is a good constructor function. The created object's (the instance's) `makeNoise` property should be a function, not an array. Actually we wanted you to give `Pet` instances a `makeNoise` method without putting a property on each object, by putting a prototype method on `Pet`. Half credit.